win32ss/user/user32/windows/mdi.c

```diff
@@ -2112,30 +2136,51 @@ CascadeWindows(HWND hwndParent, UINT wFlags, LPCRECT lpRect,
     y = rcWork.top;
     dx = GetSystemMetrics(SM_CXSIZEFRAME) + GetSystemMetrics(SM_CXSIZE);
     dy = GetSystemMetrics(SM_CYSIZEFRAME) + GetSystemMetrics(SM_CYSIZE);
+    cxWork = rcWork.right - rcWork.left;
+    cyWork = rcWork.bottom - rcWork.top;
     hwndPrev = NULL;
     for (i = info.chwnd; i > 0;)    /* in reverse order */
     {
         --i;
         hwnd = info.ahwnd[i];
 
+        if (!IsWindowVisible(hwnd) || IsIconic(hwnd))
+            continue;
+
+        if ((info.wFlags & MDITILE_SKIPDISABLED) && !IsWindowEnabled(hwnd))
+            continue;
+
         if (IsZoomed(hwnd))
             ShowWindow(hwnd, SW_RESTORE | SW_SHOWNA);
 
         GetWindowRect(hwnd, &rcWnd);
-        cx = rcWnd.right - rcWnd.left;
-        cy = rcWnd.bottom - rcWnd.top;
+        cxNew = cx = rcWnd.right - rcWnd.left;
+        cyNew = cy = rcWnd.bottom - rcWnd.top;
+
+        /* if we can change the window size and it is not only one */
+        if (info.chwnd != 1 && (GetWindowLong(hwnd, GWL_STYLE) & WS_THICKFRAME))
```

Comment:
Please use GetWindowLongPtrW.

Reply:
Usage of GetWindowLong for GWL_STYLE is correct even in Win64. There is no "GWLP_STYLE".

Reply:
Used GetWindowLongW.

Reply:
Used GetWindowLongPtrW.